### ISSUE TITLE
Checkbox/Radio when label is not next to either.. 

### ISSUE THREAD

**Issue by njohnson** [uswds/uswds#2148]
The original purpose of labels and label for was that we all used layout tables and label would be in one table cell and input in the other, yet when I separate the two even with a span or try to wrap the label around the checkbox, the styling doesn't work.    

I have an issue that a span is dynamically generated around the input but not the label..    I have always tried wrapping a label around a checkbox which also doesn't style.     Please add some flexibility around your checkboxes and radios. 

I also have the other issue  with pointers that I see others have had. and pulled the suggestion.

**Comment by AndreaSigz** [comment]
Thanks @njohnson! @maya or @donjo any thoughts?

**Comment by joshbruce** [comment]
@njohnson: Would it be possible to paste in sample HTML and a screenshot. I'm having difficulty visualizing the code you have and the result you're getting.

**Comment by njohnson** [comment]
It is a complex TOC page where the checkboxes collect data to print, create
PDF or in some cases depending on role, save to selected chapters to a file.

The mid-tier is .dotnet 4 webforms.   We are using their checkbox labeling
system.

The visual user does not need the label so I have hidden it  creating a
span within the label with the class .usa-sr-only.  The approved design
also has no visual label

  <div class="heading-row">

  <a href="#" id="" class="buttonControl hideshow">

      <span><img id="collapsed" runat="server"
src="~/Content/images/fmcsa/plus-black.svg" width="162" height="168"
alt="collapsed" >

       <span class="usa-sr-only"><%--For screenreaders only--%>click to see
subsections</span></span>

          </a>

                        <h2>

                           <a id="hypChapters" runat="server"
 href='...'></a>

                        </h2>

                        <asp:HiddenField ID="..." runat="server"
Value='...'></asp:HiddenField>

                     *   <asp:Checkbox ID="..." runat="server"
Checked='...' />         *

*                        <asp:Label ID=".." runat="server" Text="..."
AssociatedControlID="..." ><span class="usa-sr-only">check to save chapter
to my files</span></asp:Label> *

</div>

Each line has the option to expand to sub-sections, the title of the
chapter and the checkbox to collect data to print.

I've tried

1.      wrapping the label around the chapter that is a <h2 and link to the
content, but it broke the backend, so the engineer asked me to remove it

2.      wrapping the label around the input, lost the styling

3.      tried adding the .usa-sr-only to the label and finally

4.      adding a span within the label with a class .usa-sr-only.

5.      The .dotnet checkbox also generated a span around the checkbox so I
had to ask the engineer to modify this as it lost the styling.

6.      The checkbox is now picking up the styling from  _inputs.scss line
21, input,textarea, select {width:100%} so the hotspot is outside the
checkbox.

a.      Adding a div around the checkbox and label with a position relative
in the div seems to help but not perfect.


Thanks,

**Comment by joshbruce** [comment]
Hmmmmm...

I may not know enough about .NET; so, I do apologize. I'm also writing some of this for my own verification or to cover someone else coming in; not making assumptions about your aptitude, I could be wrong with some of this.

The `label` + `input` interaction (clicking the label, interacts with the input) uses the `for` + `id` attributes on the two elements. The standards seem to try and cover this possibility by making the width of the `input` 100%; thereby, as you point out, putting the hit area over the label, not just the input.

I would **recommend** changing this because it also negates the idea of adding a link to the label, like the use case I have for my own site where registering users must agree to the terms of service, which is linked to in the label:

<img width="414" alt="screen shot 2017-09-26 at 1 16 24 pm" src="https://user-images.githubusercontent.com/15252830/30873998-11ebd2a0-a2bd-11e7-8e79-8f35a0efcc28.png">

Putting `width: auto` on the checkbox/radio definition should do the trick (I haven't upgraded to the latest standards; so, mine is a little different:

```css
// Custom checkboxes

[type=checkbox],
[type=radio] {
  // The actual input element is only visible to screen readers, because
  // all visual styling is done via the label.
  @include sr-only();
  
  width: auto;

  .lt-ie9 & {
    border: 0;
    float: left;
    margin: 0.4em 0.4em 0 0;
    position: static;
    width: auto;
  }
}
```

This does seem to be an odd user experience if I'm understanding the on-screen output properly. Maybe reconsider this user flow?

The general html pattern I use for checkboxes is something like the following:

```html
<fieldset>
  <legend class="usa-sr-only">{label}</legend>
  <span class="usa-error-label">{error text}</span>
  <span class="usa-hint">{hint text}</span>
  <ul>
    <li>
      <input type="checkbox" name="input1[]" id="input1"><label for="input1">{checkbox label}</label>
    </li>
  </ul>
</fieldset>
```

At this point, you should be able to handle an error on form submission. Add hint text. Optionally display the `legend` and so on. From there, you can make it look however you like...including adding the links to the checkbox labels (which it sounds like you want to do).

I'm still not certain what the output looks like (don't share if you can't). It may be a case of the .NET labeling system not being flexible enough in its output, as opposed to the standards not being flexible enough in its definition. (Note: I do not work for 18f.)

In other words, in the example HTML above, the only part that's necessary for styling purposes, according to the standards is the `input` and the `label`. Everything else is something I do in order to create a more generic pattern for form controls and semantic markup regarding a `legend`, the error and hint texts, and so on. Wrapping in the `h2` obviously gave you something, but I'm not sure what.

Are the standards the only CSS being applied?

ps.

The more generic pattern being:

```html
<block-element>
  <label or legend></label or legend>
  <error-label></error-label>
  <hint-label></hint-label>
  <input(s)></input>
</block-element>
```

If that makes sense. If not, look at the code sample from the `text inputs`: https://standards.usa.gov/components/form-controls/#text-input

**Comment by maya** [comment]
Looks like a duplicate of #1071, closing.
